tensorflow/core/lib/strings/str_util.cc

```diff
@@ -416,6 +417,86 @@ bool ConsumeNonWhitespace(StringPiece* s, StringPiece* val) {
   }
 }
 
+Status SplitUTF8(StringPiece text, const string& delim,
```

Comment:
Isn't this function invoked only when `delim` is the empty string? 
I realize that you're perhaps adding a more general function, but would it make sense to not have this function in `str_util.h` but instead inline a function to split into UTF8 characters when the delimiter is empty in the anonymous namespace in `kernels/string_split_op.cc`?

Reply:
@asimshankar I updated the PR so that it is possible to have a non empty `delim` when encoding is `utf8`. However, because of the limitation of the current ops definition, only one char (instead of 1-4 chars in utf8 case) is allowed.